pkg/cloud/services/networking/port.go

```diff
@@ -58,6 +59,19 @@ func (s *Service) GetPortFromInstanceIP(instanceID string, ip string) ([]ports.P
 	return s.client.ListPort(portOpts)
 }
 
+type PortListOpts struct {
+	DeviceOwner []string `q:"device_owner"`
+	NetworkID   string   `q:"network_id"`
+}
```

Comment:
@pierreprinetti: I know you've been looking at this from Gophercloud perspective. Is this the way to go here?

Reply:
Eventually hopefully not, but I think this is the only way to go currently.